python/GafferSceneUI/InstancerUI.py

```diff
@@ -64,7 +64,9 @@
 			The object on which to make the instances. The
 			position, orientation and scale of the instances
 			are taken from per-vertex primitive variables on
-			this object.
+			this object. This is ignored when a filter is
+			connected, in which case the filter specifies
+			multiple objects to make the instances from.
```

Comment:
I feel like this UI is now a lot more confusing than it could have been.  Is there any real performance advantage to ever using the parent plug?  ( Now that we have the test for whether it exists, it might even be slower than just using the filter? )  Could we just standardize on always using the filter, and just have the parent plug for backwards compatibilty?  ( Then we could just default it to "", and hide it whenever it's empty ).

Reply:
This approach was chosen mainly because of the Parent node, where it makes a lot of sense. For the Parent node, `parent` defaults to "/" and is very useful that way, so removing it wasn't really an option. I could be convinced to remove it from the UI for the Instancer though, just keeping it around for backwards compatibility. I'd like to get @themissingcow and @murrays-ie's input first though, since they originally OK'd this plan.

Reply:
> For the Parent node, `parent` defaults to "/" and is very useful that way, so removing it wasn't really an option

I know that's useful currently, but its probably not intuitive to a new user (most of them ask for a merge node I think and are perplexed when we point them to Parent).

Given we want to have a `HierarchyMerge` in the near(ish) future, it'd be nice if it handled the "merge the root" efficiently enough to not need to point people back to Parent.

Also, I thought the direction we were heading was that filterable nodes without a filter attached are a no-op, so people's intuition should be that Parent needs a PathFilter to `/` before it does anything, right?

Reply:
> Given we want to have a HierarchyMerge in the near(ish) future, it'd be nice if it handled the "merge the root" efficiently enough to not need to point people back to Parent.

HierarchyMerge and Parent are different operations - HierarchyMerge isn't useable to introduce a new thing at the root unless you know for sure there isn't a thing there with the same name already.

> Also, I thought the direction we were heading was that filterable nodes without a filter attached are a no-op, so people's intuition should be that Parent needs a PathFilter to / before it does anything, right?

It is. I'm just trying to resolve that against a need for backwards compatibility and a quick parent-to-root operation. And this is what I had agreed with Murray and Tom. How about this :

- I hide the `parent` plug for all newly-created Instancer nodes.
- I keep Parent as-is, and we reconsider when/if we have an equivalent parent-to-root operation available elsewhere.

Reply:
This seems acceptable to me.  Though I don't know if we need to have an exactly equivalent parent-to-root operation elsewhere.

> HierarchyMerge and Parent are different operations 

This is true, but in the common case that we care about, either works.  Usually users know that the things they are grouping together are unique, and they just want a root level merge, and from a user perspective using a node named "Parent" is just a weird workaround for not having a merge node.

Reply:
>> - I hide the parent plug for all newly-created Instancer nodes.
>> - I keep Parent as-is,
> This seems acceptable to me.

Done in 3d4f4fa. I've hidden it for all subclasses except the Parent node.